### ISSUE TITLE
Use Debouncing in Book Search to reduce API Calls

### ISSUE THREAD

**Issue by Mrhb787** [Bookshlf-in/Website#264]
- Currently for any change in characters in Search Text Field, Search API is called.
- This increased load on server as well as adds unnecessary search results to database.

### Solution

- Adding **Debouncing** would be a great option.
- You can find out about [debouncing](https://medium.com/@jamischarles/what-is-debouncing-2505c0648ff1) here.

**Comment by YashJain2409** [comment]
Hey, can I work on this?

**Comment by Mrhb787** [comment]
Hi @YashJain2409 assigned it to you.

**Comment by Mrhb787** [comment]
Hi Yash, Seems modules are missing. 
Try deleting the node-modules folder & package-lock.json.
then again do `npm install`.

Also is problem persists share your package.json file located in root directory of project

**Comment by YashJain2409** [comment]
![book](https://user-images.githubusercontent.com/60182679/194623153-fc15ff17-a670-4987-b91d-cded1751af40.PNG)
i tried deleting node_modules and package-lock.json , it's not working

**Comment by Mrhb787** [comment]
Hi @YashJain2409,

## Method 1 - Simple
- Please update your branch with latest `origin/dev` branch.
- Again try `npm i`.

## Method 2 - Package.json Fixes
- In you package.json locate dependencies.
- Inside dependencies find dependency `"filepond-plugin-file-validate-type": "^1.2.6",`
- Add  dependency ` "filepond": "^4.30.4",`.
- e.g
```
"filepond": "^4.30.4",
"filepond-plugin-file-validate-type": "^1.2.6",
"filepond-plugin-image-exif-orientation": "^1.0.11",
"filepond-plugin-image-preview": "^4.6.10",
 ...
```
Any Method should probably fix the problem.

- I don't know but there has been a missing dependency as `"filepond": "^4.30.4"` that must be causing the issue. earlier it wasn't a required dependency hence we didn't include, but now i guess it causes breaking changes.

**Comment by YashJain2409** [comment]
hey, do i have to add debouncing wherever we are searching for results ?

**Comment by Mrhb787** [comment]
For this issue it is required only for user search panel.

**Comment by Mrhb787** [comment]
Fixed in #270 
Closing it now
